internal_worker.go

```diff
@@ -799,73 +844,49 @@ func newAggregatedWorker(
 
 	processTestTags(&wOptions, &workerParams)
 
-	env := getHostEnvironment()
-	// workflow factory.
-	var workflowWorker Worker
+	env := newHostEnvironment()
+	// workflow factory
+	var workflowWorker WorkflowWorker
 	if !wOptions.DisableWorkflowWorker {
-		if env.lenWorkflowFns() > 0 {
-			workflowFactory := newRegisteredWorkflowFactory()
-			testTags := getTestTags(wOptions.BackgroundActivityContext)
-			if testTags != nil && len(testTags) > 0 {
-				workflowWorker = newWorkflowWorkerWithPressurePoints(
-					workflowFactory,
-					service,
-					domain,
-					workerParams,
-					testTags,
-				)
-			} else {
-				workflowWorker = newWorkflowWorker(
-					getWorkflowDefinitionFactory(workflowFactory),
-					service,
-					domain,
-					workerParams,
-					nil)
-			}
+		testTags := getTestTags(wOptions.BackgroundActivityContext)
+		if testTags != nil && len(testTags) > 0 {
+			workflowWorker = newWorkflowWorkerWithPressurePoints(
+				service,
+				domain,
+				workerParams,
+				testTags,
+				env,
+			)
 		} else {
-			logger.Warn("Workflow worker is enabled but no workflow is registered. Use cadence.RegisterWorkflow() to register your workflow.")
```

Comment:
This means we can register workflows after object creation. Since Register* functions are on the object itself, we don't know yet what the workflows/activities are.

However I am planning to add a check so Register panics if called after worker.Start. That is the behavior we want to prevent.